src/Kernel.php

```diff
@@ -59,4 +72,133 @@ public function registerContainerConfiguration(LoaderInterface $loader): void
+                'pdg.config.yml',
+                'pdg.config.dist.yaml',
+                'pdg.config.dist.yml',
+            ];
+
+            foreach ($files as $filename) {
+                if (is_file(Path::join($cwd, $filename))) {
+                    $configFile = $filename;
+                    break;
+                }
+            }
+        }
+
+        // No config file detected
+        if (!$configFile) {
+            return new Configuration();
+        }
+
+        $root = Path::getDirectory($configFile);
+        
+        $normalizer = new ObjectNormalizer(nameConverter: new CamelCaseToSnakeCaseNameConverter);
+        // Config file detected: read it and parse it
+        $config = (new Processor)->process(
+            $this->getConfigTree(), 
+            Yaml::parse(file_get_contents($configFile))
+        );
+
+        if (isset($config['autoload'])) {
+            // Autoload project autoloader
+            $autoload = Path::join($root, $config['autoload']);
+            if (!file_exists($autoload)) {
+                throw new RuntimeException(sprintf('Autoload file "%s" does not exist.', $autoload));
+            }
+            require_once $autoload;
+        }
+
+        return new Configuration(autoload: $config['autoload'], guides: $normalizer->denormalize($config['guides'], Guides::class), references: $normalizer->denormalize($config['references'], References::class));
+    }
+
+    private function getConfigTree(): NodeInterface
+    {
+        $treeBuilder = new TreeBuilder('pdg');
+        $rootNode = $treeBuilder->getRootNode();
+
+        $rootNode
+            ->children()
+                ->scalarNode('autoload')
+                    ->info('Project autoload')
+                    ->defaultValue('vendor/autoload.php')
+                    ->cannotBeEmpty()
+                ->end()
+                ->arrayNode('references')
+                    ->info('References configuration')
+                    ->children()
+                        ->scalarNode('src')
+                            ->info('Root path for code parsing')
+                            ->defaultValue('src')
+                            ->cannotBeEmpty()
+                        ->end()
+                        ->scalarNode('namespace')
+                            ->info('Root namespace')
+                            ->defaultValue('App')
+                            ->cannotBeEmpty()
+                        ->end()
+                        ->arrayNode('exclude')
+                            ->info('Files or directories to ignore (supports pattern syntax)')
+                            ->scalarPrototype()->end()
+                            ->defaultValue([])
+                        ->end()
+                        ->arrayNode('class_tags_to_ignore')
+                            ->info('PHP tags to ignore')
+                            ->scalarPrototype()->end()
+                            ->defaultValue(['@internal', '@experimental'])
```

Comment:
```suggestion
                            ->defaultValue(['@internal', '@experimental', '@ignore'])
```
https://docs.phpdoc.org/guide/references/phpdoc/tags/ignore.html